src/components/date-time-input/date-time-input.ts

```diff
@@ -617,6 +625,10 @@ export default class IgcDateTimeInputComponent extends EventEmitterMixin<
       this.updateMask();
     }
 
+    if (this._oldValue !== this.value) {
+      this.handleChange();
+    }
```

Comment:
I reckon part of the logic in the blur handler can be refactored into the "default" change handler, specifically the partially completed mask handling.
Anyway, I'm ok with the current implementation and this can be addressed in a separate PR since it would be an internal change